src/step_definitions/elements.js

```diff
@@ -20,58 +20,58 @@ defineSupportCode(function ({ When, Then }) {
   });
 
   When(/^I click the "([^"]*)" element$/, function (elementName) {
-    const self = this;
-
-    return self.currentPage.scrollIntoElement(elementName)
-      .then(function () {
-        return self.currentPage.click(elementName)
-          .then(
-            null,
-            function () {
-              console.warn('Warning! Element was not clickable. We need to scroll it down.');
-              return browser.executeScript('window.scrollBy(0,50);')
-                .then(function () {
-                  return self.currentPage.click(elementName)
-                    .then(
-                      null,
-                      function () {
-                        return Promise.reject(`Error, after scrolling the element "${elementName}" is still not clickable.`);
-                      }
-                    );
-                });
-            }
-          );
+    return this.currentPage.waitForVisibilityOf(elementName)
+      .then(() => {
+        this.currentPage.scrollIntoElement(elementName);
+      })
+      .then(() => {
+        this.currentPage.click(elementName);
+      })
+      .catch(error => {
+        console.warn('Warning! Element was not clickable. We need to scroll it down.');
+        return browser.executeScript('window.scrollBy(0,50);')
+          .then(() => this.currentPage.click(elementName));
+      }).catch(error => {
+        return Promise.reject(`Error, after scrolling the element "${elementName}" is still not clickable.`);
       });
+
   });
 
   When(/^I click the "([^"]*)" "([^"]*)" element$/, function (elementName, parameter) {
-    const self = this;
-
-    return self.currentPage.scrollIntoElement(elementName)
-      .then(function () {
-        return self.currentPage[elementName](parameter).click();
+    return this.currentPage.waitForVisibilityOf(elementName)
+      .then(() => {
+        this.currentPage.scrollIntoElement(elementName);
+      })
+      .then(() => {
+        this.currentPage[elementName](parameter).click();
       });
   });
 
   When(/^I click the "([^"]*)" element if it is visible$/, function (elementName) {
-    const self = this;
-
-    return this.currentPage.isVisible(elementName).then(function () {
-      return self.currentPage.scrollIntoElement(elementName)
-          .then(function () {
-            return self.currentPage.click(elementName);
+    return this.currentPage.isVisible(elementName)
+      .then(() => {
```

Comment:
There is a huge difference between `() => this.currentPage.scrollIntoElement(elementName)` and 
`() => { this.currentPage.scrollIntoElement(elementName) }`.

Your current code won't work at all, because you don't return a promise. That was the reason I asked for a tests for the steps you're changing.